torch_stoi/stoi.py

```diff
@@ -116,70 +116,116 @@ def forward(self, est_targets: torch.Tensor,
                 est_targets.view(-1, wav_len),
                 targets.view(-1, wav_len),
             ).view(inner)
+        batch_size = targets.shape[0]
+
         if self.do_resample and self.sample_rate != FS:
             targets = self.resample(targets)
             est_targets = self.resample(est_targets)
 
+        if self.use_vad:
+            targets, est_targets, mask = self.remove_silent_frames(
+                targets, est_targets, self.dyn_range, self.win_len,
+                self.win_len//2
+            )
+            # Remove the last mask frame to replicate a bug in pystoi
+            # The bug is caused by a matlab-like indexing which removes a frame
+            # See https://github.com/mpariente/pystoi/issues/32
+            mask, _ = mask.sort(-1, descending=True)
+            mask = mask[..., 1:]
+
         # Here comes the real computation, take STFT
         x_spec = self.stft(targets, self.win, self.nfft, overlap=2)
         y_spec = self.stft(est_targets, self.win, self.nfft, overlap=2)
+        # Reapply the mask because of overlap in STFT
+        if self.use_vad:
+            x_spec *= mask.unsqueeze(1)
+            y_spec *= mask.unsqueeze(1)
+
+        """Uncommenting the following lines and the last block at the end
+        allows to replicate the pystoi behavior when less than N speech frames
+        are detected
+        """
+        # # Ensure at least 30 frames for intermediate intelligibility
+        # if self.use_vad:
+        #     not_enough = mask.sum(-1) < self.intel_frames
+        #     if not_enough.any():
+        #         import warnings
+        #         warnings.warn('Not enough STFT frames to compute intermediate '
+        #                       'intelligibility measure after removing silent '
+        #                       'frames. Returning 1e-5. Please check you wav '
+        #                       'files', RuntimeWarning)
+        #         if not_enough.all():
+        #             return torch.full(batch_size, 1e-5)
+        #         x_spec = x_spec[~not_enough]
+        #         y_spec = y_spec[~not_enough]
+        #         mask = mask[~not_enough]
```

Comment:
Why do you want to comment this ?

Reply:
Was not sure if we want to have this behavior for `torch_stoi`. Let me know if you think we should, and if we don't if I should remove the comments.

Reply:
To be clear I did not just comment this, I implemented it and then commented it.

Reply:
Ok, let's keep it like this (commented).